test/Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -342,32 +346,57 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                             "Transfer-Encoding: chunked",
                             "",
                             "gg");
-                        await connection.ReceiveForcedEnd(
-                            "HTTP/1.1 400 Bad Request",
-                            "Connection: close",
-                            $"Date: {server.Context.DateHeaderValue}",
-                            "Content-Length: 0",
-                            "",
-                            "");
+                        if (expectedClientStatusCode == HttpStatusCode.OK)
```

Comment:
Do expectedClientStatusCode and expectedServerStatusCode always match now?

Reply:
No. Some tests set null vs 0, and one sets null vs 400.